### PR TITLE
Feat: App listing dropdown api filter added for appname

### PR DESCRIPTION
**Pull request by vikramdevtron**
# Description

Filter added in app listing by app name as wildcard. appName param is optional

Fixes # (issue) : #1739 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] autocomplete api with and without appName filter.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have tested it for all user roles
* [ ] I have added all the required unit/api test cases

### CODE REVIEW

File: internal/sql/repository/app/AppRepository.go [hunk 1]
```diff
@@ -236,3 +237,9 @@ func (repo AppRepositoryImpl) FindAppAndProjectByAppName(appName string) (*App,
 		Select()
 	return app, err
 }
+
+func (repo AppRepositoryImpl) FindAllMatchesByAppName(appName string) ([]App, error) {
```
Comment (nishant-d) [inline internal/sql/repository/app/AppRepository.go:241]: use pointer

File: pkg/pipeline/PipelineBuilder.go [hunk 2]
```diff
@@ -2330,3 +2331,22 @@ func (impl PipelineBuilderImpl) GetCiPipelineById(pipelineId int) (ciPipeline *b
 	ciPipeline.PostBuildStage = postStageDetail
 	return ciPipeline, err
 }
+
+func (impl PipelineBuilderImpl) FindAllMatchesByAppName(appName string) ([]AppBean, error) {
+	var appsRes []AppBean
+	var apps []app2.App
```
Comment (nishant-d) [inline pkg/pipeline/PipelineBuilder.go:2337]: use pointers

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: internal/sql/repository/app/AppRepository.go (hunk 1 above)
Comment (vikramdevtron) [reply internal/sql/repository/app/AppRepository.go]: fixed

File: pkg/pipeline/PipelineBuilder.go (hunk 2 above)
Comment (vikramdevtron) [reply pkg/pipeline/PipelineBuilder.go]: fixed

**Review by nishant-d — APPROVED** [review]

**Pull request merged**